addons/hr_holidays/static/src/bugfix/bugfix.js

```diff
@@ -8,3 +8,81 @@ odoo.define('hr_holidays/static/src/bugfix/bugfix.js', function (require) {
 'use strict';
 
 });
+
+// FIXME move me in hr_holidays/static/src/models/partner/partner.js
+odoo.define('hr_holidays/static/src/models/partner/partner.js', function (require) {
+'use strict';
+
+const {
+    registerClassPatchModel,
+    registerFieldPatchModel,
+    registerInstancePatchModel,
+} = require('mail/static/src/model/model_core.js');
+const { attr, one2one } = require('mail/static/src/model/model_field.js');
+const { clear } = require('mail/static/src/model/model_field_command.js');
+
+const { str_to_datetime } = require('web.time');
+
+registerClassPatchModel('mail.partner', 'hr_holidays/static/src/models/partner/partner.js', {
+    /**
+     * @override
+     */
+    convertData(data) {
+        const data2 = this._super(data);
+        if ('out_of_office_date_end' in data) {
+            data2.out_of_office_date_end = data.out_of_office_date_end ? data.out_of_office_date_end : clear();
+        }
+        return data2;
+    },
+});
+
+registerInstancePatchModel('mail.partner', 'hr_holidays/static/src/models/partner/partner.js', {
+    /**
+     * @private
+     */
+    _computeOutOfOfficeText() {
+        if (!this.out_of_office_date_end) {
+            return clear();
+        }
+        if (!this.messaging.locale.language) {
+            return clear();
+        }
+        const currentDate = new Date();
+        const date = str_to_datetime(this.out_of_office_date_end);
+        const options = { day: 'numeric', month: 'short' };
+        if (currentDate.getFullYear() !== date.getFullYear()) {
+            options.year = 'numeric';
+        }
+        const localeCode = this.messaging.locale.language.replace(/_/g,'-');
```

Comment:
Prefer using `this.env.messaging` than `this.messaging`, for consistency in how to access `messaging`. Having `messaging` as related field is a workaround with dependencies, but we will eventually no longer need dependencies for almost all computes.